runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,66 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview`, or `open` commands depending on
+--- the operating system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---
+---@param opts table|nil Optional keyword arguments:
+---             - confirm boolean|nil Ask user for confirmation before opening.
+---
+---@return string|nil returns the string output of the command. nil if
+---opts.confirm and the user declined. See |system()|.
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri, opts)
+  vim.validate({
+    opts = { opts, 'table', true },
+  })
+
+  opts = opts or {}
+  local cmd
+  local sysname = vim.loop.os_uname().sysname
+
+  if sysname == 'Darwin' then
+    cmd = 'open'
+  elseif sysname:match('windows') then
+    cmd = 'wslview'
+  else -- Assume linux
+    cmd = 'xdg-open'
+  end
+
+  if opts.confirm then
+    local answer =
+      vim.fn.confirm('Open "' .. uri .. '" using ' .. cmd .. '?', '&Yes\n&No', 2, 'Question')
+
+    if answer == 0 or answer == 2 then
+      return
+    end
+  end
```

Comment:
I'd leave this out of this PR as it is a behavior change for LSP and it's not clear to me that having this confirm dialog is an improvement.

And it also seems to me that there's no good reason to build  this check into `open_uri` instead of doing it outside.